### ISSUE TITLE
NPE if Clojure isn't on classpath

### ISSUE THREAD

**Issue by ajoberstar** [projectodd/shimdandy#6]
If Clojure isn't on the classpath when `init()` runs, an NPE occurs. This seems to be because `postInvoke()` assumes `dvals` is set, but when Clojure isn't on the classpath this doesn't occur.

The only _issue_ here is just the confusing error message that occurs, since the original error that presumably has a ClassNotFound or something is buried by the NPE.

```
Caused by: java.lang.NullPointerException
        at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.postInvoke(ClojureRuntimeShimImpl.java:49)
        at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.init(ClojureRuntimeShimImpl.java:31)
        at org.projectodd.shimdandy.ClojureRuntimeShim.newRuntime(ClojureRuntimeShim.java:26)
        at gradle_clojure.plugin.internal.ClojureWorker$ClojureRuntime.get(ClojureWorker.java:76)
        at gradle_clojure.plugin.internal.ClojureWorker.run(ClojureWorker.java:31)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:39)
        at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:46)
        at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:30)
        at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:101)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:155)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:137)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
        ... 3 more
```

**Comment by tobias** [comment]
Yeah, it seems like we should see a `ClassNotFoundException` first, since we get access to `dvals` via `clojure.lang.Var`, and I suspect we are, but since `postInvoke` gets called in the `finally` block, the NPE gets thrown first. I'll add a null check in `postInvoke` to handle that case.

**Comment by tobias** [comment]
This is fixed in 1.2.1, which should be syncing to Maven Central sometime in the next hour or so. Thanks for the report!

**Comment by ajoberstar** [comment]
Thanks for the quick fix!
